ogre2/src/Ogre2ThermalCamera.cc

```diff
@@ -263,18 +295,14 @@ void Ogre2ThermalCameraMaterialSwitcher::preRenderTargetUpdate(
           for (unsigned int i = 0; i < item->getNumSubItems(); ++i)
           {
             Ogre::SubItem *subItem = item->getSubItem(i);
-            if (!subItem->hasCustomParameter(this->customParamIdx))
-            {
-              // normalize temperature value
-              float color = temp * 100.0 /
-                  static_cast<float>(std::numeric_limits<uint16_t>::max());
-
-              // set g, b, a to 0. This will be used by shaders to determine
-              // if particular fragment is a heat source or not
-              // see media/materials/programs/thermal_camera_fs.glsl
-              subItem->setCustomParameter(this->customParamIdx,
-                  Ogre::Vector4(color, 0, 0, 0.0));
-            }
+            // normalize temperature value
+            float color = (temp / this->resolution) / ((1 << bitDepth) - 1.0);
+
+            // set g, b, a to 0. This will be used by shaders to determine
+            // if particular fragment is a heat source or not
+            // see media/materials/programs/thermal_camera_fs.glsl
+            subItem->setCustomParameter(this->customParamIdx,
```

Comment:
Do we no longer need the `if (!subItem->hasCustomParameter(this->customParamIdx))` check?

Reply:
the value of the `color` variable above is different for an 8 bit and a 16 bit camera. So when testing in a world containing both 16 bit and 8 bit cameras, we'll need to constantly override the values on each camera update.